ext/search_kit/Civi/Api4/Action/SearchDisplay/GetSearchTasks.php

```diff
@@ -138,15 +138,30 @@ public function _run(\Civi\Api4\Generic\Result $result) {
             $task['title'] = E::ts('Profile Update');
           }
           $key = \CRM_Core_Key::get(\CRM_Utils_Array::first((array) $task['class']), TRUE);
-          $tasks[$entity['name']]['contact.' . $id] = [
-            'title' => $task['title'],
-            'icon' => $task['icon'] ?? 'fa-gear',
-            'crmPopup' => [
-              'path' => "'{$task['url']}'",
-              'query' => "{reset: 1}",
-              'data' => "{cids: ids.join(','), qfKey: '$key'}",
-            ],
-          ];
+
+          //break out mailing labels for redirect instead of crmPopup
+          if ($id == CRM_Core_Task::LABEL_CONTACTS) {
+            $tasks[$entity['name']]['contact.' . $id] = [
+              'title' => $task['title'],
+              'icon' => $task['icon'] ?? 'fa-gear',
+              'redirect' => [
+                'path' => "'{$task['url']}'",
+                'query' => "{reset: 1}",
+                'data' => "{cids: ids.join(','), qfKey: '$key'}",
+              ],
+            ];
+          }
+          else {
+            $tasks[$entity['name']]['contact.' . $id] = [
+              'title' => $task['title'],
+              'icon' => $task['icon'] ?? 'fa-gear',
+              'crmPopup' => [
```

Comment:
It appears the only difference between these 2 blocks of code is this one line 'crmPopup' vs. 'redirect'. That could be a variable. Like

```
// Print Labels action does not support popups, open full-screen
$actionType = $id == CRM_Core_Task::LABEL_CONTACTS ? 'redirect' : 'crmPopup';
```

Reply:
That does make for a smaller more elegant code change, thanks @colemanw